sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -388,10 +388,9 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
     )
   }
 
-  def mergeUnsupportedByWindowFunctionError(): SparkUnsupportedOperationException = {
-    new SparkUnsupportedOperationException(
-      errorClass = "_LEGACY_ERROR_TEMP_2010",
-      messageParameters = Map.empty)
+  def mergeUnsupportedByWindowFunctionError(): Throwable = {
+    SparkException.internalError(
+      s"""Aggregate Window Functions do not support merging.""")
```

Comment:
Could you use " instead of """
and please, add the function name. How about:
```suggestion
      s"The aggregate window function ${toSQLId(funcName)} does not support merging.")
```

Pass the function name:
```scala
  override lazy val mergeExpressions =
    throw QueryExecutionErrors.mergeUnsupportedByWindowFunctionError(prettyName)
```